src/dowel/logger.py

```diff
@@ -206,6 +209,8 @@ def log(self, data):
             if isinstance(data, output.types_accepted):
                 output.record(data, prefix=self._prefix_str)
                 at_least_one_logged = True
+        if at_least_one_logged and isinstance(data, TabularInput):
```

Comment:
Good catch that this needs to be done here. It's definitely a weakness of the tabular API.